### ISSUE TITLE
ci.coords should fail more gracefully with ret = "threshold"

### ISSUE THREAD

**Issue by xrobin** [xrobin/pROC#24]
I believe the only supported use case of boostrapping threshold is with 'x = "best"'. For all other cases, pROC should produce a useful error message, not garbage like:

```
> ci.coords(roc1, x=0.8, input = "sensitivity", ret=c("specificity", "ppv", "tp", "thr"))
Error in apply(sapply(perfs, c), 1, quantile, probs = c(0 + (1 - conf.level)/2,  : 
  dim(X) must have a positive length
De plus : Warning message:
In ci.coords.roc(roc1, x = 0.8, input = "sensitivity", ret = c("specificity",  :
  NA value(s) produced during bootstrap were ignored.
```
or
```
> ci.coords(roc1, x=0.9, input = "sensitivity", ret="t")
95% CI (2000 stratified bootstrap replicates):
                           2.5% 50% 97.5%
sensitivity 0.9: threshold   NA  NA    NA
Warning message:
In ci.coords.roc(roc1, x = 0.9, input = "sensitivity", ret = "t") :
  NA value(s) produced during bootstrap were ignored.
```

In the longer term, work should continue on [branch interpolate](https://github.com/xrobin/pROC/tree/interpolate) that will ultimately support this feature by interpolating thresholds.

**Comment by xrobin** [comment]
Fixed in c298256f7d2dba7ce1a1c49542e03e4e2e6da2c1.

**Comment by xrobin** [comment]
The `interpolate` branch is outdated and was archived as tag [`archive/interpolate`](https://github.com/xrobin/pROC/tree/archive/interpolate).
